config.go

```diff
@@ -233,6 +235,10 @@ type Config struct {
 	// latest features. Setting it to a version greater than you are actually
 	// running may lead to random breakage.
 	Version KafkaVersion
+	// The registry to define metrics into.
+	// Defaults to metrics.DefaultRegistry.
+	// See Examples on how to use the metrics registry
+	MetricRegistry metrics.Registry
```

Comment:
I think it must not be null as it would panic when invoking methods on it.
Although we could add `if ... != nil` conditions when using the registry I think the official way is to set `metrics.UseNilMetrics` to `true`:
https://godoc.org/github.com/rcrowley/go-metrics#pkg-variables

We can have that setting the default but I remember the overhead being minimal during my tests and when you need high performance you generally wants the metrics to capture those performance.
The Java client has metrics turn on by default too.

I can update some benchmarks to see the impact of metrics being enabled vs disabled.